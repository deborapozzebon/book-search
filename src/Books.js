import React, { useState } from 'react';

class Books extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.books !== null && this.props.books !== undefined && this.props.books !== '') {
            return (
                <div className="row">
                    <div className="col-xs-6 col-md-3">
                        {this.props.books.map(function (book, i) {
                            return (
                                <a href="#" class="thumbnail">
                                    <img src={book.volumeInfo.imageLinks.thumbnail} alt={book.title} />
                                </a>
                            )
                        })}
                    </div>
                </div>
            )
        }
        return null;
    }
}

export default Books;