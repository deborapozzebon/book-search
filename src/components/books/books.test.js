import Books from './books';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Books></Books>, div);
    ReactDOM.unmountComponentAtNode(div);
})

test('snapshot 1', () => {
    const book = [{
        title: 'JavaScript',
        Description: 'JavaScript book to begginers'
    }]
    const tree = renderer.create(<Books books={book}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
