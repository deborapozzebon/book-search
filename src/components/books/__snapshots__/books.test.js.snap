// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1 1`] = `
<div
  className="row"
>
  <div
    className="col-sm"
  >
    <a
      className="bk-thumb"
      onClick={[Function]}
    >
      <img
        alt="JavaScript"
      />
    </a>
  </div>
</div>
`;
